### ISSUE TITLE
Property Wrapper cannot be compiled correctly with `$` in function body (Xcode 14 and main?)

### ISSUE THREAD

**Issue by freddi-kit** [swiftlang/swift#59295]
**Describe the bug**
The code with Property Wrapper cannot be compiled correctly in Swift 5.7 dev (Xcode 14 dev)

```swift
import Foundation

@propertyWrapper
struct WrapperValue<Value> {
  var value: Value
  init(wrappedValue: Value) {
    self.value = wrappedValue
  }

  var projectedValue: Self {
    return self
  }
  
  var wrappedValue: Value {
    get {
      self.value  
    }
    set {
      self.value = newValue
    }
  }

  func printValue() {
    print(value)
  }
}


class Test {
  static func test() {
    (0..<100).map { _ in
      @WrapperValue var value: Bool = false

      if value {
        $value.printValue() // .. compile error on 5.7
      }
    }
  }
}

```

This code had been compiled until Swift 5.6.1.

I already send it to apple (FB10059129)

**To Reproduce**
Compile above code.

**Expected behavior**
Compile success.

**Screenshots**

```
<stdin>:35:9: error: cannot reference invalid declaration '$value'
        $value.printValue() // .. compile error on 5.7
        ^
<stdin>:32:25: note: '$value' declared here
      @WrapperValue var value: Bool = false
                        ^
```

**Environment (please complete the following information):**
 - Xcode 14 dev (14A5228q)
 - nightly (ee312bc1e20eb01)

**Comment by freddi-kit** [comment]
Hello, this stil fails nightly build, could anyone help checking it?

**Comment by hank121314** [comment]
Smaller reproducer:

```swift
@propertyWrapper
struct WrapperValue {
  var value: Bool
  init(wrappedValue: Bool) {
    self.value = wrappedValue
  }

  var wrappedValue: Bool {
    get {
      self.value
    }
    set {
      self.value = newValue
    }
  }
}

let _ =  { () -> () in
  @WrapperValue var value = false

  _value
}
```

**Comment by freddi-kit** [comment]
In Swift 5.7, it is not fixed yet. Please check with the code example in this issue's description @xedin

**Comment by freddi-kit** [comment]
```
$ swift --version
swift-driver version: 1.62.8 Apple Swift version 5.7 (swiftlang-5.7.0.127.4 clang-1400.0.29.50)
Target: arm64-apple-macosx12.0
```

**Comment by freddi-kit** [comment]
it looks fixed on Xcode 14.1 RC, thanks
